### PR TITLE
add vision_estimation message for vision based position and

### PR DESCRIPTION
**Pull request by mrivi**
velocity estimation

### CODE REVIEW

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -4356,5 +4356,25 @@
...
+      <field type="uint8_t" name="estimator_type" enum="MAV_ESTIMATOR_TYPE">Class id of the estimator this estimate originated from.</field>
+      <field type="float" name="x" units="m">X Position</field>
+      <field type="float" name="y" units="m">Y Position</field>
+      <field type="float" name="z" units="m">Z Position</field>
+      <field type="float" name="q1" units="rad">Quaternion component 1, x</field>
+      <field type="float" name="q2" units="rad">Quaternion component 2, y</field>
+      <field type="float" name="q3" units="rad">Quaternion component 3, z</field>
+      <field type="float" name="q4" units="rad">Quaternion component 4, w</field>
+      <field type="float[36]" name="pose_covariance">Covariance matrix (first six entries are the first ROW, next six entries are the second row, etc.)</field>
+      <field type="float" name="vx" units="m/s">X Linear Velocity</field>
```
Comment (julianoes) [inline message_definitions/v1.0/common.xml]: I think we should specify NED or body or whatever the frame is.

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -4356,5 +4356,25 @@
...
+      <description>Vision based position and velocity estimation</description>
+      <field type="uint64_t" name="time_usec" units="us">Timestamp (microseconds since system boot or since UNIX epoch)</field>
+      <field type="uint8_t" name="estimator_type" enum="MAV_ESTIMATOR_TYPE">Class id of the estimator this estimate originated from.</field>
+      <field type="float" name="x" units="m">X Position</field>
+      <field type="float" name="y" units="m">Y Position</field>
+      <field type="float" name="z" units="m">Z Position</field>
+      <field type="float" name="q1" units="rad">Quaternion component 1, x</field>
+      <field type="float" name="q2" units="rad">Quaternion component 2, y</field>
+      <field type="float" name="q3" units="rad">Quaternion component 3, z</field>
+      <field type="float" name="q4" units="rad">Quaternion component 4, w</field>
```
Comment (julianoes) [inline message_definitions/v1.0/common.xml]: I would use `float [4]` like in other some other newer messages.

**Comment by TSC21** [comment]
@mrivi I don't think we require this PR, for 4 reasons:
1. You have [`VISION_POSITION_ESTIMATE`](http://mavlink.org/messages/common#VISION_POSITION_ESTIMATE) - we can though extend it to use quaternions;
2 . You have [`VISION_SPEED_ESTIMATE`](http://mavlink.org/messages/common#VISION_SPEED_ESTIMATE);
3. I already proposed https://github.com/mavlink/mavlink/pull/724, which extends the msgs to add the respective covariance matrices;
4. Having two separate msgs for pose and for velocity is a better approach as the payload will be much lower and not everyone is using both position and velocity estimates.

**Comment by mrivi** [comment]
closing in favor of #785

**Pull request closed without merging**